crates/doc/src/ser.rs

```diff
@@ -7,15 +7,23 @@ use std::io;
 pub struct SerPolicy {
     /// Truncate strings which are longer than this limit.
     pub str_truncate_after: usize,
+    pub array_truncate_after: usize,
+    pub obj_truncate_after: usize,
+    pub nested_obj_truncate_after: usize,
 }
 
 impl SerPolicy {
-    pub fn new(str_truncate_after: usize) -> Self {
-        Self { str_truncate_after }
+    pub fn truncate_strings(str_truncate_after: usize) -> Self {
```

Comment:
what about a builder pattern? `new()` returns a SerPolicy that does no pruning, and then you call `with_str_truncate_after` or `with_array_truncate_after` etc.

I'm guessing `obj_truncate_after` is in the policy just to make root => node transitions easier to track, right?